packages/client/src/features/move-zoom/zoom-handler.ts

```diff
@@ -0,0 +1,158 @@
+/********************************************************************************
+ * Copyright (c) 2023 Business Informatics Group (TU Wien) and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { Action, Bounds, Point, SetViewportAction, Viewport } from '@eclipse-glsp/protocol';
+import { inject, injectable } from 'inversify';
+import { throttle } from 'lodash';
+import {
+    findParentByFeature,
+    IActionDispatcher,
+    IActionHandler,
+    ICommand,
+    isViewport,
+    SChildElement,
+    SModelElement,
+    SModelRoot
+} from 'sprotty';
+import { EditorContextService } from '../../base/editor-context-service';
+import { TYPES } from '../../base/types';
+import { getElements, isSelectableAndBoundsAware, SelectableBoundsAware } from '../../utils/smodel-util';
+
+export interface ZoomViewportAction extends Action {
+    kind: typeof ZoomViewportAction.KIND;
+    zoomFactor: number;
+}
+
+export namespace ZoomViewportAction {
+    export const KIND = 'zoomViewportAction';
+
+    export function is(object: any): object is ZoomViewportAction {
+        return Action.hasKind(object, KIND);
+    }
+
+    export function create(zoomFactor: number): ZoomViewportAction {
+        return { kind: KIND, zoomFactor };
+    }
+}
+export interface ZoomElementAction extends Action {
+    kind: typeof ZoomElementAction.KIND;
+    elementIds: string[];
+    zoomFactor: number;
+}
+
+export namespace ZoomElementAction {
+    export const KIND = 'zoomElementAction';
+
+    export function is(object: any): object is ZoomElementAction {
+        return Action.hasKind(object, KIND);
+    }
+
+    export function create(elementIds: string[], zoomFactor: number): ZoomElementAction {
+        return { kind: KIND, elementIds, zoomFactor };
+    }
+}
+
+@injectable()
+export class ZoomViewportHandler implements IActionHandler {
+    @inject(EditorContextService)
+    protected editorContextService: EditorContextService;
+
+    static readonly defaultZoomInFactor = 1.1;
+    static readonly defaultZoomOutFactor = 0.9;
+
+    @inject(TYPES.IActionDispatcher) protected dispatcher: IActionDispatcher;
+    protected readonly throttledHandleViewportZoom = throttle((action: ZoomViewportAction) => this.handleZoomViewport(action), 500);
```

Comment:
Thanks for this fix!
It feels better now, but could we use a lower timeout here?
I tested with 150, this also feel less laggy. wdyt?